server/src/test/java/org/opensearch/index/mapper/DenseVectorMapperTests.java

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.mapper;
+
+import org.opensearch.common.Strings;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.index.mapper.DenseVectorFieldMapper.DenseVectorFieldType;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.opensearch.index.mapper.KnnAlgorithmContext.Method.HNSW;
+import static org.opensearch.index.mapper.KnnAlgorithmContextFactory.HNSW_PARAMETER_BEAM_WIDTH;
+import static org.opensearch.index.mapper.KnnAlgorithmContextFactory.HNSW_PARAMETER_MAX_CONNECTIONS;
+
+public class DenseVectorMapperTests extends MapperServiceTestCase {
```

Comment:
```suggestion
public class DenseVectorFieldMapperTests extends FieldMapperTestCase2<DenseVectorFieldMapper.Builder> {
```

We should use the `FieldMapperTestCase2` helper class here to ensure we covering merge conflic and serialization tests that are important for the REST client interfaces.

For guidance on this test see the [LegacyGeoShapeFieldMapperTests](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/index/mapper/LegacyGeoShapeFieldMapperTests.java#L68) example

Reply:
I can't use ```FieldMapperTestCase2``` directly as ```DenseVectorFieldMapper.Builder``` extends [ParametrizedFieldMapper.Builder](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/ParametrizedFieldMapper.java#L582) that is not supported by ```FieldMapperTestCase2```, seems that majority of other mapper builders extend ```ParametrizedFieldMapper.Builder``` as well. 

@nknize what would be your suggestion, is it better to keep new mapper for sense_vector as ```ParametrizedFieldMapper.Builder``` and write custom test cases for merge conflicts, or change it to ```FieldMapper.Builder```? I do not see any strong argument for keeping it as ```ParametrizedFieldMapper.Builder``` except that it will require additional efforts to re-write current implementation, while with ```FieldMapper.Builder``` we obviously will have merge and serialization tests for free?

Reply:
I've done refactoring, but to b honest I'm not sure if that is better than a previous version. While test coverage is better, there is a lot of additional code added that has been taking care of by the ```ParametrizedFieldMapper```. Please check and let me know what do you think. Thx!